geniza/entities/models.py

```diff
@@ -146,6 +151,168 @@ def __str__(self):
         except Name.DoesNotExist:
             return str(self.names.first() or super().__str__())
 
+    def merge_with(self, merge_people, user=None):
+        """Merge the specified people into this one. Combines all metadata
+        into this person and creates a log entry documenting the merge.
+
+        Closely adapted from :class:`Document` merge."""
+
+        # if user is not specified, log entry will be associated with script
+        if user is None:
+            user = User.objects.get(username=settings.SCRIPT_USERNAME)
+
+        # language codes are needed to merge description, which is translated
+        language_codes = [lang_code for lang_code, lang_name in settings.LANGUAGES]
+
+        # handle translated description: create a dict of descriptions
+        # per supported language to aggregate and merge
+        description_chunks = {
+            lang_code: [getattr(self, "description_%s" % lang_code)]
+            for lang_code in language_codes
+        }
+
+        # collect names as strings (since that's the important part for comparison)
+        self_names = [str(name) for name in self.names.all()]
+
+        for person in merge_people:
+            # ensure any has_page overrides are respected
+            if person.has_page and not self.has_page:
+                self.has_page = True
+
+            # ensure missing role and gender are migrated
+            if person.role and not self.role:
+                self.role = person.role
+            if self.gender == Person.UNKNOWN and person.gender != Person.UNKNOWN:
+                self.gender = person.gender
```

Comment:
am I understanding correctly, these are only set if not already present? (comment could be slightly clearer, maybe something like: migrate/copy role and gender if not already present)

does it matter if there's a conflict?

Reply:
I think it's probably worth raising a validation error if there's a conflict, just to make sure researchers don't make a mistake and combine people who shouldn't actually be merged.